### ISSUE TITLE
Rule proposal: `no-http`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#971]
Enforce using HTTPS URLs over HTTP.

## Fail

```js
const foo = 'http://sindresorhus.com';

// Hello: http://sindresorhus.com
```

## Pass

```js
const foo = 'https://sindresorhus.com';

// Hello: https://sindresorhus.com
```

---

Alternative name: `prefer-https`. Which one do people prefer?

I'm aware this could be partly solved with the `string-content` rule, but I don't think most people would think of this and it also wouldn't work in comments.

**Comment by sindresorhus** [comment]
This is now accepted. The rule name should be `prefer-https`.

**Comment by himynameisdave** [comment]
I can pick this up!

**Comment by JounQin** [comment]
Should `http://localhost` for example be ignored?

**Comment by sindresorhus** [comment]
> Should http://localhost for example be ignored?

Yes, any URL without a TLD should be ignored.

**Comment by manovotny** [comment]
Is this still desired since [string-content](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/string-content.md) can accomplish this now?

**Comment by sindresorhus** [comment]
Yes, that's addressed in the issue description.

**Comment by manovotny** [comment]
My apologies @sindresorhus. That's what I get for skimming too quickly. 😞

**Comment by manovotny** [comment]
FWIW, just in case it helps here, this is the regex I came up with for use with `string-content`.

```
(?!(?=.*(localhost|0\.0\.0\.0|127\.0\.0\.1)))^http\:
```

Demo here on regex101: https://regex101.com/r/ITZChx/1

**Comment by sindresorhus** [comment]
If anyone wants to work on this, see the initial attempt and feedback in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1084.

**Comment by lxl66566** [comment]
<img width="1496" height="603" alt="Image" src="https://github.com/user-attachments/assets/c22a112c-8b09-4eb5-83ef-5480b3b547f3" />

When running `eslint --fix` on my Tauri v2 project, this rule rewrites the `http` scheme to `https`. This causes all images in the app to fail loading with the error: `Failed to load resource: net::ERR_CONNECTION_REFUSED`.

I would hope to more carefully consider whether this rule should be enabled by default or whether its auto‑fix should be applied.

**Comment by himynameisdave** [comment]
> this rule rewrites the http scheme to https.

That's the point. You can disable the rule in your own local config if you don't like it.

Consider serving your images from https.
